### ISSUE TITLE
Unicode in the cli

### ISSUE THREAD

**Issue by ftyers** [MycroftAI/mycroft-core#1140]
It doesn't seem to be possible to type Unicode in the Mycroft CLI... I tried both "Qué hora es?" and "Сколько времени?" and neither worked. It seems only possible to type ASCII.

**Comment by forslund** [comment]
Feel free to test PR #1273, not 100% sure about the implementation but works for me (TM)

**Comment by forslund** [comment]
#1273 was merged so some Unicode is now supported in the cli. I'm closing this issue but I'd be happy for reports if there's still issues with this.

**Comment by zeehio** [comment]
I still am having issues with unicode in the cli.

This line of code (and the following) are to blame:

https://github.com/MycroftAI/mycroft-core/blob/1dee7fc381ad8579540ef314576ae8ac7ff630fe/mycroft/client/text/main.py#L896

`getch()` does not handle unicode. See:
 -  [`getch` documentation](https://docs.python.org/2/library/curses.html#curses.window.getch)
 - [Python2 bug: `getch` does not support unicode](https://bugs.python.org/issue18118)
 - [`get_wch` documentation (python3.3)](https://docs.python.org/3.3/library/curses.html#curses.window.get_wch)

This limits my ability to easily test translated skills using the cli, without needing to setup a wake word, speech recognition, speech synthesis...

Looks like this issue is blocked by: https://github.com/MycroftAI/mycroft-core/issues/366

**Comment by neterbg** [comment]
I'm working about translating Mycroft in bulgarian. For the purpose, I made these changes in my mycroft-core/mycroft/client/text/main.py:

Added this on new row after "import sys":
`reload(sys)`
`sys.setdefaultencoding('utf8')`

Changed this:
`return text.encode('utf8')`
to this:
`return text`

Changed this:
`line = line[:-1]`
to this:
`line = line.decode('utf8')[:-1]`
`line = line.encode('utf8')`

Changed this:
`elif curses.ascii.isascii(c):`
to this:
`else:`

I don't send it as pull request because I saw many opinions in Internet that using of sys.setdefaultencoding() is discouraged and I'm not sure if all about these changes is right and optimal, but it works for me, running it on Debian Stretch with python 2.7.13, I can read messages and write commands without problems in cyrillic (bulgarian) in the cli.

I'm still not sure if it is needed, but I added the first snippet in the __init__.py files of the skills (plus the row "import sys" if there is no such in the file). Plus I added this as first row of these files:
`# coding=utf8`
This helps if you write/import some non-latin stuff in these files.
